### ISSUE TITLE
Make File Error In MacOS High Sierra 10.13.2 

### ISSUE THREAD

**Issue by achillesss** [litecoin-project/litecoin#412]
### Describe the issue
make file got error.
### Can you reliably reproduce the issue?
yes
#### If so, please list the steps to reproduce below:
1. ``make`` after ``./configure`` under ``/litecoin``

### Expected behaviour
Shouldn't it return no error ?

### Actual behaviour
Logs: 

```bash
Making all in src
  CXX      libbitcoin_server_a-miner.o
In file included from miner.cpp:6:
In file included from ./miner.h:10:
In file included from ./txmempool.h:25:
In file included from /usr/local/include/boost/multi_index/ordered_index.hpp:17:
/usr/local/include/boost/multi_index/detail/ord_index_impl.hpp:1181:10: error: no matching function for call to object of type 'const key_compare'
      (aka 'const CompareModifiedEntry')
      if(comp_(key(v),key(y->value())))return false;
         ^~~~~
/usr/local/include/boost/multi_index/detail/ord_index_impl.hpp:825:9: note: in instantiation of member function
      'boost::multi_index::detail::ordered_index_impl<boost::multi_index::identity<CTxMemPoolModifiedEntry>, CompareModifiedEntry,
      boost::multi_index::detail::nth_layer<2, CTxMemPoolModifiedEntry, boost::multi_index::indexed_by<boost::multi_index::ordered_unique<modifiedentry_iter,
      CompareCTxMemPoolIter, mpl_::na>, boost::multi_index::ordered_non_unique<boost::multi_index::tag<ancestor_score, mpl_::na, mpl_::na, mpl_::na, mpl_::na,
      mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na>,
      boost::multi_index::identity<CTxMemPoolModifiedEntry>, CompareModifiedEntry>, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na,
      mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na>, std::__1::allocator<CTxMemPoolModifiedEntry>
      >, boost::mpl::v_item<ancestor_score, boost::mpl::vector0<mpl_::na>, 0>, boost::multi_index::detail::ordered_non_unique_tag,
      boost::multi_index::detail::null_augment_policy>::in_place' requested here
      b=in_place(x->value(),x,Category());
        ^
/usr/local/include/boost/multi_index/detail/ord_index_impl.hpp:860:18: note: in instantiation of member function
      'boost::multi_index::detail::ordered_index_impl<boost::multi_index::identity<CTxMemPoolModifiedEntry>, CompareModifiedEntry,
      boost::multi_index::detail::nth_layer<2, CTxMemPoolModifiedEntry, boost::multi_index::indexed_by<boost::multi_index::ordered_unique<modifiedentry_iter,
      CompareCTxMemPoolIter, mpl_::na>, boost::multi_index::ordered_non_unique<boost::multi_index::tag<ancestor_score, mpl_::na, mpl_::na, mpl_::na, mpl_::na,
      mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na>,
      boost::multi_index::identity<CTxMemPoolModifiedEntry>, CompareModifiedEntry>, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na,
      mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na>, std::__1::allocator<CTxMemPoolModifiedEntry>
      >, boost::mpl::v_item<ancestor_score, boost::mpl::vector0<mpl_::na>, 0>, boost::multi_index::detail::ordered_non_unique_tag,
      boost::multi_index::detail::null_augment_policy>::modify_' requested here
      if(!super::modify_(x)){
                 ^
/usr/local/include/boost/multi_index_container.hpp:823:18: note: in instantiation of member function
      'boost::multi_index::detail::ordered_index_impl<modifiedentry_iter, CompareCTxMemPoolIter, boost::multi_index::detail::nth_layer<1,
      CTxMemPoolModifiedEntry, boost::multi_index::indexed_by<boost::multi_index::ordered_unique<modifiedentry_iter, CompareCTxMemPoolIter, mpl_::na>,
      boost::multi_index::ordered_non_unique<boost::multi_index::tag<ancestor_score, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na,
      mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na>,
      boost::multi_index::identity<CTxMemPoolModifiedEntry>, CompareModifiedEntry>, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na,
      mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na>, std::__1::allocator<CTxMemPoolModifiedEntry>
      >, boost::mpl::vector0<mpl_::na>, boost::multi_index::detail::ordered_unique_tag, boost::multi_index::detail::null_augment_policy>::modify_' requested here
      if(!super::modify_(x)){
                 ^
/usr/local/include/boost/multi_index/detail/index_base.hpp:278:21: note: in instantiation of function template specialization
      'boost::multi_index::multi_index_container<CTxMemPoolModifiedEntry, boost::multi_index::indexed_by<boost::multi_index::ordered_unique<modifiedentry_iter,
      CompareCTxMemPoolIter, mpl_::na>, boost::multi_index::ordered_non_unique<boost::multi_index::tag<ancestor_score, mpl_::na, mpl_::na, mpl_::na, mpl_::na,
      mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na>,
      boost::multi_index::identity<CTxMemPoolModifiedEntry>, CompareModifiedEntry>, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na,
      mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na>, std::__1::allocator<CTxMemPoolModifiedEntry>
      >::modify_<update_for_parent_inclusion>' requested here
    {return final().modify_(mod,x);}
                    ^
/usr/local/include/boost/multi_index/detail/ord_index_impl.hpp:418:18: note: in instantiation of function template specialization
      'boost::multi_index::detail::index_base<CTxMemPoolModifiedEntry, boost::multi_index::indexed_by<boost::multi_index::ordered_unique<modifiedentry_iter,
      CompareCTxMemPoolIter, mpl_::na>, boost::multi_index::ordered_non_unique<boost::multi_index::tag<ancestor_score, mpl_::na, mpl_::na, mpl_::na, mpl_::na,
      mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na>,
      boost::multi_index::identity<CTxMemPoolModifiedEntry>, CompareModifiedEntry>, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na,
      mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na>, std::__1::allocator<CTxMemPoolModifiedEntry>
      >::final_modify_<update_for_parent_inclusion>' requested here
    return this->final_modify_(
                 ^
miner.cpp:373:31: note: in instantiation of function template specialization 'boost::multi_index::detail::ordered_index_impl<modifiedentry_iter,
      CompareCTxMemPoolIter, boost::multi_index::detail::nth_layer<1, CTxMemPoolModifiedEntry,
      boost::multi_index::indexed_by<boost::multi_index::ordered_unique<modifiedentry_iter, CompareCTxMemPoolIter, mpl_::na>,
      boost::multi_index::ordered_non_unique<boost::multi_index::tag<ancestor_score, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na,
      mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na>,
      boost::multi_index::identity<CTxMemPoolModifiedEntry>, CompareModifiedEntry>, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na,
      mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na>, std::__1::allocator<CTxMemPoolModifiedEntry>
      >, boost::mpl::vector0<mpl_::na>, boost::multi_index::detail::ordered_unique_tag,
      boost::multi_index::detail::null_augment_policy>::modify<update_for_parent_inclusion>' requested here
                mapModifiedTx.modify(mit, update_for_parent_inclusion(it));
                              ^
./miner.h:76:10: note: candidate function not viable: 'this' argument has type 'const key_compare' (aka 'const CompareModifiedEntry'), but method is not marked
      const
    bool operator()(const CTxMemPoolModifiedEntry &a, const CTxMemPoolModifiedEntry &b)
         ^
In file included from miner.cpp:6:
In file included from ./miner.h:10:
In file included from ./txmempool.h:25:
In file included from /usr/local/include/boost/multi_index/ordered_index.hpp:17:
/usr/local/include/boost/multi_index/detail/ord_index_impl.hpp:1186:26: error: no matching function for call to object of type 'const key_compare'
      (aka 'const CompareModifiedEntry')
    return y==header()||!comp_(key(y->value()),key(v));
                         ^~~~~
./miner.h:76:10: note: candidate function not viable: 'this' argument has type 'const key_compare' (aka 'const CompareModifiedEntry'), but method is not marked
      const
    bool operator()(const CTxMemPoolModifiedEntry &a, const CTxMemPoolModifiedEntry &b)
         ^
2 errors generated.
make[2]: *** [libbitcoin_server_a-miner.o] Error 1
make[1]: *** [all-recursive] Error 1
make: *** [all-recursive] Error 1
```

### What version of litecoin-core are you using?
(HEAD detached at v0.14.2)

### Machine specs:
- OS: MacOS 10.13.2
- CPU: 2.5 GHz Intel Core i7
- RAM: 16 GB 1600 MHz DDR3
- Disk size: 500GB
- Disk Type (HD/SDD): SDD

**Comment by santos88** [comment]
I can confirm the same error with MacOS High Sierra.

**Comment by ydeleon6** [comment]
I just finished building the application under MacOS Sierra 10.12, no error encountered. Looks like this might be limited to your newer os.

**Comment by jbeich** [comment]
Maybe cherry-pick bitcoin/bitcoin@1ec0c0a01c31.

**Comment by ydeleon6** [comment]
Ok!

**Comment by achillesss** [comment]
@santos88 yes, High Sierra performs this error very well 😄

**Comment by AlMel3000** [comment]
I confrim issue with MacOS  ‎10.13.2

**Comment by VKoskiv** [comment]
Can confirm that cherry-picking https://github.com/bitcoin/bitcoin/commit/1ec0c0a01c31 resolves this issue on macOS 10.13.2

**Comment by Akiqqqqqqq** [comment]
try v.16.0
